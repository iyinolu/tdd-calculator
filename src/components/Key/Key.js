import React from "react";
import PropTypes from "prop-types";

const Key = ({ keyAction, keyType, keyValue }) => {
  return (
    <div className="key-container">
      <p className="key-value">{keyValue}</p>
    </div>
  );
};

Key.propTypes = {
  keyValue: PropTypes.string.isRequired,
  keyAction: PropTypes.func.isRequired,
  keyType: PropTypes.string.isRequired,
};

Key.defaultProps = {
  keyValue: "",
  keyType: "number",
  keyAction: () => {},
};

export default Key;
