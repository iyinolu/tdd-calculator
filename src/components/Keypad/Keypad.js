import React from "react";
import PropTypes from "prop-types";
import Key from "../Key/Key";
import "../../styles/Keypad.css"

const Keypad = ({
  callOperator,
  numbers,
  operators,
  setOperator,
  updateDisplay,
}) => {
  const renderValues = (values) =>
    values.map((number, key) => <p key={key}>{number}</p>);

  return (
    <div className="keypad-container">
      <div className="numbers-container">{renderValues(numbers)}</div>
      <div className="operators-container">{renderValues(operators)}</div>
      <Key />
    </div>
  );
};

Keypad.propTypes = {
  callOperator: PropTypes.func.isRequired,
  numbers: PropTypes.array.isRequired,
  operators: PropTypes.array.isRequired,
  setOperator: PropTypes.func.isRequired,
  updateDisplay: PropTypes.func.isRequired,
};

export default Keypad;
