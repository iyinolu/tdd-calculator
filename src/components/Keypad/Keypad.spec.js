import React from 'react';
import { shallow } from 'enzyme';
import Keypad from "./Keypad";
import Key from "../Key/Key";

describe("Keypad", () => {
    let wrapper;
    beforeEach(() => wrapper = shallow(<Keypad 
        callOperator={jest.fn()}
        numbers={[]}
        operators={[]}
        setOperator={jest.fn()}
        updateDisplay={jest.fn()}
    />))

    it("should contain 3 divs", () => {
        expect(wrapper.find("div").length).toEqual(3)
    })

    it("should render the values of numbers props", () => {
        wrapper.setProps({numbers: ["1", "2", "3"]})
        expect(wrapper.find(".numbers-container").text()).toEqual("123")
    })

    it("should render the values of operator props", () => {
        wrapper.setProps({operators: ['+', '-', '*', '/']})
        expect(wrapper.find(".operators-container").text()).toEqual("+-*/")
    })

    it("should render Key component", () => {
        expect(wrapper.containsMatchingElement(<Key />)).toEqual(true)
    })

    it("should contain one instance of Key component", () => {
        expect(wrapper.find('Key').length).toEqual(1)
    })
})