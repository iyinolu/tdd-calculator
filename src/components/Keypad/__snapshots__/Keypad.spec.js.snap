// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad should render properly 1`] = `ShallowWrapper {}`;
